src/ncp/rcp_host.cpp

```diff
@@ -213,7 +167,15 @@ void RcpHost::Init(void)
 
     VerifyOrExit(otLoggingSetLevel(level) == OT_ERROR_NONE, error = OTBR_ERROR_OPENTHREAD);
 
-    mInstance = otSysInit(&mConfig);
+    // There are two cases:
+    // 1. The `Init` is called after `CreateInstance`. In this case, `otSysInit` has been called
+    // and an otInstance has been created. So there is no need to create an instance again.
+    // 2. `Reset` is called. `Deinit` is called first and then `Init` is called again. In this
+    // case need to call `otSysInit` to reset the co-processor and creat an otInstance again.
+    if (mInstance == nullptr)
+    {
+        mInstance = otSysInit(&mConfig);
+    }
```

Comment:
It looks error-prone that we have to create the otInstance at different places. Probably we should look for a better approach.

Reply:
I agree. 

The difficulty here is that `otSysInit` actually contains 3 steps:
1. `platformInit`
2. create otInstance 
3. `platformSetUp`

And now we do need `platformInit` to run at the very beginning so that we know the co-processor type. However we do not have access to `platformInit`. We can only call `otSysInit` which includes creating the otInstance.

When we do something like `Reset`. We also need to call `otSysInit` again. This caused the problem we are facing now.

I think we can refactor `otSysInit` and extract `platformSpinelManagerInit` out of it as a new posix otSys Api. And at the beginning we only call `platformSpinelManagerInit` to know the co-processor type. Then `otSysInit` will still be called later. It will also try to call `platformSpinelManagerInit` but it finds that it has been initialized and won't initialize it again. Thoughts? @superwhd

Reply:
I have submitted a change in openthread: https://github.com/openthread/openthread/pull/10344

With this change, we don't need to call `otSysInit` and create otInstance when trying to determine the co-processor type. Please take a look at that PR first.